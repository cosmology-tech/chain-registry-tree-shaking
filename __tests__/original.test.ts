import { AssetList, Chain } from '@chain-registry/types';
import {
    assets,
    chains
} from 'chain-registry';

import type { PartialAssetList, PartialChain } from '../src';

it('Chain type', () => {
    // just to check types..
    const myChain: Chain = chains[0];
    expect(myChain.chain_id).toMatchSnapshot();
})

it('Chain simplified snap', () => {
    const myChain: PartialChain = {
        chain_name: chains[0].chain_name,
        chain_id: chains[0].chain_id,
        bech32_prefix: chains[0].bech32_prefix,
        slip44: chains[0].slip44,
        staking: chains[0].staking
    }
    expect(myChain).toMatchSnapshot();
})

it('AssetList type', () => {
    // just to check types..
    const myAssetList: AssetList = assets[0];
    expect(myAssetList.chain_name).toMatchSnapshot();
})

it('AssetList simplified snap', () => {
    const myAssetList: PartialAssetList = {
        chain_name: assets[0].chain_name,
        assets: assets[0].assets
    }
    expect(myAssetList).toMatchSnapshot();
})