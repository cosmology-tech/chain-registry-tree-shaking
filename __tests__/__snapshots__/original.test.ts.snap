// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetList simplified snap 1`] = `
{
  "assets": [
    {
      "base": "uebl",
      "coingecko_id": "",
      "denom_units": [
        {
          "denom": "uebl",
          "exponent": 0,
        },
        {
          "denom": "ebl",
          "exponent": 6,
        },
      ],
      "description": "The native staking token of 8ball.",
      "display": "ebl",
      "images": [
        {
          "png": "https://raw.githubusercontent.com/cosmos/chain-registry/master/8ball/images/8ball.png",
          "svg": "https://raw.githubusercontent.com/cosmos/chain-registry/master/8ball/images/8ball.svg",
        },
      ],
      "logo_URIs": {
        "png": "https://raw.githubusercontent.com/cosmos/chain-registry/master/8ball/images/8ball.png",
        "svg": "https://raw.githubusercontent.com/cosmos/chain-registry/master/8ball/images/8ball.svg",
      },
      "name": "8ball",
      "symbol": "EBL",
    },
  ],
  "chain_name": "8ball",
}
`;

exports[`AssetList type 1`] = `"8ball"`;

exports[`Chain simplified snap 1`] = `
{
  "bech32_prefix": "8ball",
  "chain_id": "eightball-1",
  "chain_name": "8ball",
  "slip44": 118,
  "staking": {
    "staking_tokens": [
      {
        "denom": "uebl",
      },
    ],
  },
}
`;

exports[`Chain type 1`] = `"eightball-1"`;
